File: src/tenwordswisdom.py
t = int(input())
for _ in range(t):
    n = int(input())
    idx = 0; q = 0
    for i in range(1, n + 1):
        a, b = map(int, input().split())
        if b > q and a <= 10: 
            q = b; idx = i
    print(idx)

File: src/tenwordswisdom.cpp
#include<iostream>
using namespace std;

int main(){
    int t, n, a, b; cin >> t;
    while(t--){
        cin >> n;
        int idx, q = 0;
        for(int i = 1; i <= n; i++){
            cin >> a >> b;
            if((b > q) && (a <= 10)){
                q = b; idx = i;
            }
        }
        cout << idx << endl;
    }
    return 0;
}